appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/boot/runtime/MicroGlassFish.java

```diff
@@ -74,19 +75,20 @@ public void start() throws GlassFishException {
     }
 
     @Override
-    public void stop() throws GlassFishException {
+    public synchronized void stop() throws GlassFishException {
         if (status == Status.STOPPED || status == Status.STOPPING || status == Status.DISPOSED) {
             throw new IllegalStateException("Already in " + status + " state.");
         }
-        
+
         status = Status.STOPPING;
```

Comment:
Question is: after status is `STOPPING` will this prevent accepting new requests? I think the intention is to finish the ongoing ones but I guess we do not want to accept new ones.

Reply:
I don't follow what the question is here?
There's a check literally above this that prints an error if multiple things try to stop the server.

Reply:
What I wanted to point out is that before the wait only the `status` was changed. No internal processing has been signalled to stop by that point. I would imagine that it could very well be the case that new request still are accepted and start to process. If the goal with this feature is to give time to finish started requests it seams only consequent not to accept new ones. I would assume that some parts of the server needs to be stopped or flagged to get this behaviour.

Reply:
Searching the source I don't see that status is used much other than to print it into the admin console. Perhaps we could use this status variable better to stop processing etc.However that would be a different JIRA.